clang/lib/3C/CastPlacement.cpp

```diff
@@ -183,7 +183,12 @@ void CastPlacementVisitor::surroundByCast(ConstraintVariable *Dst,
   // of adding a new expression.
   if (auto *CE = dyn_cast<CStyleCastExpr>(E->IgnoreParens())) {
     SourceRange CastTypeRange(CE->getLParenLoc(), CE->getRParenLoc());
-    rewriteSourceRange(Writer, CastTypeRange, CastStrs.first.substr(1));
+    assert("Cast expected to start with '('" && !CastStrs.first.empty() &&
+           CastStrs.first[0] == '(');
+    std::string CastStr = CastStrs.first.substr(1);
+    bool Failed = Writer.ReplaceText(CastTypeRange, CastStr);
```

Comment:
Might there be any advantage to using `rewriteSourceRange` but somehow enhancing it so you can generate a different diagnostic on failure?